src/hotspot/share/services/virtualMemoryTracker.cpp

```diff
@@ -523,29 +525,72 @@ bool VirtualMemoryTracker::remove_released_region(address addr, size_t size) {
     }
   }
 
-  VirtualMemorySummary::record_released_memory(size, reserved_rgn->flag());
-
-  assert(reserved_rgn->contain_region(addr, size), "Not completely contained");
-  if (reserved_rgn->base() == addr ||
-      reserved_rgn->end() == addr + size) {
+  if (size <= reserved_rgn->size() && reserved_rgn->contain_region(addr, size)) {
```

Comment:
A region which contains another must necessarily have at least the size of the contained region, therefore the size check is unnecessary.

Reply:
Ah, I see now, you are correct. I will fix it.

Reply:
Done.